src/datatypes/src/vectors.rs

```diff
@@ -174,8 +174,19 @@ pub trait MutableVector: Send + Sync {
 
     /// Push value ref to this mutable vector.
     ///
-    /// Returns error if data types mismatch.
-    fn push_value_ref(&mut self, value: ValueRef) -> Result<()>;
+    /// # Panics
+    /// Panics if error if data types mismatch.
+    fn try_push_value_ref(&mut self, value: ValueRef) -> Result<()>;
+
+    /// Push value ref to this mutable vector.
+    fn push_value_ref(&mut self, value: ValueRef) {
+        self.try_push_value_ref(value).unwrap()
+    }
+
+    // Push null value ref to this mutable vector.
+    fn push_null(&mut self) -> Result<()> {
+        self.try_push_value_ref(ValueRef::Null)
+    }
```

Comment:
Since our vectors are nullable, `push_null` doesn't need to return `Result`. We could require all vector builders to implement this method
```suggestion
    fn push_null(&mut self);
```

e.g.
```rust
fn push_null(&mut self) {
    self.mutable_array.append_null()
}
```

Reply:
Thank you Yingwen for the review and suggestions 🙏
Will address them today.

Reply:
@evenyag I would like to confirm if I got the idea correct. 

1. > doesn't need to return Result

Does that mean that we can drop the `context`, so 
`vector.push_null().context(CreateVectorSnafu)`
simply becomes `vector.push_null()`

2. For builders that do not have  `mutable_array` I plan to make the following
 - for `NullVectorBuilder`
```rust
    fn push_null(&mut self) {
        self.try_push_value_ref(ValueRef::Null).unwrap();
    }
```
- for `ListVectorBuilder`.
```rust
    fn push_null(&mut self) {
        self.push_null();
    }
```
Does it look ok for you?

Reply:
>Does that mean that we can drop the context, so
>vector.push_null().context(CreateVectorSnafu)
>simply becomes vector.push_null()

Yes, since it's infallible now.

>for NullVectorBuilder

This might report a "must_use Result" warning. You can unwrap here or use `push_value_ref`.

>for ListVectorBuilder.

Looks good.